shell.nix

```diff
@@ -1,20 +1,29 @@
-# This file defines a Nix shell environment. It is normally used in conjunction
-# with a `.envrc` file and the direnv tool. But it can also be manually
-# activated by runnin `nix-shell` in this directory. Once activated, programs
-# such as ghci will be in scope.
+let
 
-{ pkgs ? import <nixpkgs> {} }:
-  pkgs.mkShell {
-    buildInputs = [
-      pkgs.ghcid
-      (pkgs.haskellPackages.ghcWithPackages (hpkgs: [
-        hpkgs.hlint
-        hpkgs.servant
-        hpkgs.servant-auth
-        hpkgs.servant-auth-server
-        hpkgs.servant-blaze
-        hpkgs.servant-server
-        hpkgs.wai
-      ]))
-    ];
+  sources = import ./nix/sources.nix;
+  nixpkgsF = import sources.nixpkgs;
+  overlays = import ./overlays.nix;
+
+  # Overlays let us override certain packages at a central location.
+  nixpkgs-overlayed = nixpkgsF { inherit overlays; };
+  nixpkgs = nixpkgsF { };
+  hp = nixpkgs-overlayed.haskellPackages;
+
+  # Brittany, as the formatter, is just here as an example.
+  # I personally prefer to have the formatters pinned to a version and then
+  # made available via direnv to avoid unnecessary diff pollution across upgrades.
+  brittany =
+    hp.callCabal2nix "brittany" sources.brittany { };
+
+  formatters = [ brittany ]; # Add more as we need them.
+
+  # Niv is great at pinning dependencies in sources.json and computing SHA's etc.
+  nix-tooling = with hp; [ niv ];
+
+  # Haskell tools
+  haskell-tooling = with hp; [ cabal-install ghcid hlint ];
+
+in hp.shellFor {
+  packages = p: with p; [ start-servant ];
+  buildInputs = haskell-tooling ++ formatters ++ nix-tooling;
```

Comment:
I would reorder the 3 lists either here, or reorder their appearance above, so they match.